purchase_request_group_by_date/models/stock_rule.py

```diff
@@ -0,0 +1,98 @@
+# Copyright 2018-2020 ForgeFlow, S.L.
+# License LGPL-3.0 or later (https://www.gnu.org/licenses/lgpl-3.0)
+
+from datetime import datetime
+
+from odoo import api, fields, models
+
+
+class StockRule(models.Model):
+    _inherit = "stock.rule"
+
+    company_group_purchase_request = fields.Boolean(
+        related="company_id.group_purchase_request"
+    )
+
+    # override to change returned domain based on company settings
+    @api.model
+    def _make_pr_get_domain(self, values):
+        domain = super()._make_pr_get_domain(values)
+        if self.company_group_purchase_request:
+            domain = (
+                ("picking_type_id", "=", self.picking_type_id.id),
+                ("company_id", "=", values["company_id"].id),
+                ("date_start", "=", datetime.today().date()),
+            )
+            gpo = self.group_propagation_option
+            group_id = (
+                (gpo == "fixed" and self.group_id.id)
+                or (gpo == "propagate" and values["group_id"].id)
+                or False
+            )
+            if group_id:
+                domain += (("group_id", "=", group_id),)
```

Comment:
better append in the domain here, so if the `_make_pr_get_domain` changes, we don't have to modify this module as well.
```suggestion
            domain += ("date_start", "=", datetime.today().date())
            gpo = self.group_propagation_option
            group_id = (
                (gpo == "fixed" and self.group_id.id)
                or (gpo == "propagate" and values["group_id"].id)
                or False
            )
            if group_id:
                domain += (("group_id", "=", group_id),)
```